### ISSUE TITLE
"Mismatched Quarkus version found" warning

### ISSUE THREAD

**Issue by nevenr** [quarkiverse/quarkus-operator-sdk#969]
Hi,

On "https://code.quarkus.io" I created simple project stub including only quarkus-operator-sdk extension. [code-with-quarkus.zip](https://github.com/user-attachments/files/17189041/code-with-quarkus.zip)

Maven build (mvn clean package) produce warning message: 
```
[WARNING] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] Mismatched Quarkus version found: "3.15.1", expected: "3.14.2" by at least a minor version and things might not work as expected.
```
When "quarkus-operator-sdk" dependency is removed warning is gone. 

More details:
```
[INFO] --- quarkus:3.15.1:build (default) @ code-with-quarkus ---
[INFO] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] QOSDK: 6.8.0 (commit: 79457f5) built on Mon Sep 09 21:42:27 CEST 2024
[INFO] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] JOSDK: 4.9.4 (commit: 3588780) built on Thu Aug 29 18:52:57 CEST 2024
[INFO] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] Fabric8 (effective): 6.13.3
[WARNING] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] Mismatched Quarkus version found: "3.15.1", expected: "3.14.2" by at least a minor version and things might not work as expected.
[INFO] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] Fabric8 (JOSDK-defined): 6.13.3
[INFO] [io.quarkiverse.operatorsdk.deployment.VersionAlignmentCheckingStep] Fabric8 (Quarkus): 6.13.3
```
Maven and java info:
```
C:\tmp\code-with-quarkus>c:\Programs\apache-maven-3.9.5\bin\mvn --version
Apache Maven 3.9.5 (57804ffe001d7215b5e7bcb531cf83df38f93546)
Maven home: c:\Programs\apache-maven-3.9.5
Java version: 21.0.4, vendor: Eclipse Adoptium, runtime: c:\Users\nevenr\.jdks\temurin-21.0.4
Default locale: en_US, platform encoding: UTF-8
OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"

C:\tmp\code-with-quarkus>
```

Regards,
N.

**Comment by metacosm** [comment]
Hi @nevenr, this is expected at the moment because the platform currently includes 6.8.0, which was build against Quarkus 3.14, and should have been included in the 3.15 platform, so seeing this warning is "normal" and shouldn't be a problem. Please don't hesitate to open issues if you find that this isn't the case.

**Comment by nevenr** [comment]
Hi @metacosm ,
All is OK. Thanks for response.
Regards,
N.
